src/scripting/lua_map.cc

```diff
@@ -2178,96 +2201,116 @@ int LuaMapObjectDescription::get_name(lua_State* L) {
+          all bobs,
+        * :class:`critter <CritterDescription>`, animals that aren't
+          controlled by any tribe,
+        * :class:`ship <ShipDescription>`, a sea-going vessel
+          belonging to a tribe that can ferry wares or an expedition,
+        * :class:`worker <WorkerDescription>`, a worker belonging to
+          a tribe,
+        * :class:`carrier <CarrierDescription>`, a specialized
+          worker for carrying items along a road,
+        * :class:`soldier <SoldierDescription>`, a specialized worker
+          that will fight for its tribe.
+
+      * **Wares:** :class:`ware <WareDescription>`, a ware used by
+        buildings to produce other wares, workers or ships
+      * **Immovables:** Immovables are map objects that have a fixed
+        position on the map, like buildings or trees. Immovable types are:
+
+        * :class:`immovable <ImmovableDescription>` General immovables
+          that can belong to a tribe (e.g. a wheat field) or to the
+          world (e.g. trees or rocks).
+
+        * **Buildings:** Buildings always belong to a tribe. Building
+          types are:
+
+          * :class:`building <BuildingDescription>`, the base class
+            for all buildings
+          * :class:`constructionsite <ConstructionSiteDescription>`,
+            an actual building is being constructed here,
+          * :class:`dismantlesite <DismantleSiteDescription>`, an
+            actual building is being dismantled here,
+          * :class:`warehouse <WarehouseDescription>`, a warehouse
+            can store wares and workers. Headquarters and ports are
+            special tapes of warehouses, but they belong to the same
+            class,
+          * :class:`militarysite <MilitarySiteDescription>`, a
+            building manned by soldiers to expand a tribe's territory,
+          * :class:`productionsite <ProductionSiteDescription>`, the
+            most common type of building, which can produce wares,
+          * :class:`trainingsite <TrainingSiteDescription>`, a
+            specialized productionsite for improving soldiers.
+
+        * **Other Immovables:** Specialized immovables that aren't buildings:
+
+          * :class:`flag <FlagDescription>`, a flag that can hold
+            wares for transport,
+          * :class:`road <RoadDescription>`, a road or waterway
+            connecting two flags,
+          * :class:`portdock <PortdockDescription>`, a 'parking space'
+            on water terrain where ships can load/unload wares and
+            workers. A portdock is invisible to the player and one is
+            automatically placed next to each port building.
+
+      * **Abstract:** These types are abstract map objects that are used by the engine and are
+        not visible on the map.
+
+        * :class:`battle <BattleDescription>`, holds information
+          about two soldiers in a fight,
+        * :class:`fleet <FleetDescription>`, holds information for
```

Comment:
Please double-check if this is still correct. There are now classes ShipFleet and FerryFleet and I don't know how the Lua interface for these looks now.

Reply:
There are no Lua interfaces about `fleet` or `ShipFleet` or `'FerryFleet`. The original documentation says `Fleet` and `BattleDescription` are 'abstract'.

Reply:
The corresponding strings are "ferry_fleet" and "ship_fleet" (defined in logic/map_objects/map_object_type.cc). Please add a note that no properties are currently implemented and remove the (probably broken) class links.

Also, for roads, the map object type strings are "roadbase" (abstract generic road/waterway), "road", and "waterway" (three different strings even though they're represented by the one and the same Lua class!), please update that as well.

Reply:
I have to think about this... my first impression is: This documentation is for Lua-scripting. Not for explaining the widelands Lua api, which should be part of the c++ documentation.

Reply:
The info about the roads and waterways needs to go here because the information stated is plainly wrong (incorrect typename strings). For fleets and battles you could also just write a brief note like

> Ferry fleets ("ferry_fleet"), ship fleets ("ship_fleet"), and battles ("battle") are also map objects, but they can not be accessed from Lua currently.

Reply:
I feel not be able to do it correctly. Please feel free to make the changes you think would be more correct.

Reply:
I have pushed a [commit](https://github.com/widelands/widelands/commit/60dde177adf4a655fe9f81fd211eec287138cd68) to update this and also the dangers of `double`. Please check if you agree with my changes.